### PR TITLE
Checkout: Add Jetpack Redirect after Jetpack Cloud Checkout[Revised]

### PR DESCRIPTION
**Pull request by elliottprogrammer**
### Changes proposed in this Pull Request

This PR sets the post checkout redirect location to an explicit Jetpack Redirect API source if we're checking out on Jetpack Cloud and it's a Jetpack product. Using the Jetpack Redirect API allows us to gracefully update the post purchase redirect location when needed.

This PR is another approach to #49425 and in my opinion is a much simpler and optimal solution.

### Testing instructions

- Checkout and run this PR in Jetpack Cloud  (`yarn start-jetpack-cloud`)
- Select a Jetpack site and go to the pricing page: `jetpack.cloud.localhost:3000/pricing/:site`
- Purchase any Jetpack product.
- Verify you are redirected to `https://cloud.jetpack.com/landing/:site?product=:productSlug&thank-you=true`
    - note: /landing redirects to /backup (which is expected for now)
- Bonus for testing various products.
- Run unit tests and verify they pass.
    -  `yarn test-client client/my-sites/checkout/composite-checkout/test/composite-checkout-thank-you.js`

Fixes: 1199409969959876-as-1199670278480660

### CODE REVIEW

**Comment by elliottprogrammer** [comment]
Still need to update unit tests..

**Review by sirbrillig — APPROVED** [review]
The code looks good to me! I haven't run through the tests but it is good from a checkout perspective.

**Review by mattgawarecki — APPROVED** [review]
All tests passed for me, and the code changes look sound. I was a little surprised not to see a `beforeEach` method to reset all the mock implementations before each test, but it wasn't here before and everything seems to work, so 🤷 

Approved, ready to go when you are! (Is the [DO NOT MERGE] tag still relevant?)

**Comment by elliottprogrammer** [comment]
Awesome @mattgawarecki , thanks for the review 👍.

> Is the [DO NOT MERGE] tag still relevant?

I think so. I could be mistaken, but I don't think we have decided yet exactly the final location we want users to land after purchase, and what post purchase behaviors should happen after purchase. There was mention of redirecting to the Jetpack Assistant project or something, but truth is, I'm not really sure. I'll work on finding out the answer. 👍

**Comment by elliottprogrammer** [comment]
Since this PR only affects checkout if `isJetpackCloud()`, this PR can be merged because Jetpack Cloud checkout is not live in prod yet (it is in staging), so merging this will not affect any customers until Jetpack Cloud Checkout is live in prod.
So merging this now. 👍

**Pull request merged**
